test/native/setup.js

```diff
@@ -156,5 +164,14 @@ jest.mock(
 		announceForAccessibility: jest.fn(),
 		removeEventListener: jest.fn(),
 		isScreenReaderEnabled: jest.fn( () => Promise.resolve( false ) ),
+		fetch: jest.fn( () => ( {
+			done: jest.fn(),
+		} ) ),
 	} )
 );
+
+// Helpers for integration tests
+global.getEditorHtml = () => {
+	triggerHtmlSerialization();
+	return serializedHtml;
+};
```

Comment:
This global helper returns the current HTML of the editor, when it's called, it simulates what we do when we switch to HTML mode.

https://github.com/WordPress/gutenberg/blob/f3a576314c647465ef5ffac8eb6cf292cda91059/packages/editor/src/components/provider/index.native.js#L85-L87

https://github.com/WordPress/gutenberg/blob/f3a576314c647465ef5ffac8eb6cf292cda91059/packages/editor/src/components/provider/index.native.js#L230-L235

**Here is an use example:**
```
describe( 'Paragraph block', () => {
	it( 'returns HTML', async () => {
		const initialHtml = `<!-- wp:paragraph -->
<p>Lorem ipsum</p>
<!-- /wp:paragraph -->`;

		await initializeEditor( { initialHtml } );
		expect( global.getEditorHtml() ).toBe( initialHtml );
	} );
} );
```

Reply:
Is verifying the final block HTML the ultimate output to validate that our app is working as expected? I am not attempting to imply that is not, I just wanted to pose the question. 

The alternative output I think we might verify if the _UI rendered on the screen_, rather than the block HTML. It may be that asserting against both the block HTML and the render UI are valid, valuable approaches in different contexts. I'm not sure. 

Additionally, does it make sense or is it required to make this helper a global or could be make it a module that is imported into test files? I hesitate creating globals unless it is necessary or provides substantial value. 

WDYT?

Reply:
>Is verifying the final block HTML the ultimate output to validate that our app is working as expected?

I thought about this more. I do think there is concrete value in asserting the outputted block HTML. We likely will write many tests that assert against both the outputted block HTML _and_ the rendered UI, as both are important aspects of the product we are building.

The only alternative approach I thought of is that we might separately unit test the pieces that manage HTML output, but I don't have a clear vision for what that might look like. I think we should take the approach testing both HTML output and UI rendered in these integration tests as you have.

Reply:
I finally decided to move this part to the helper file instead of having it in the setup file, now it's located [here](https://github.com/WordPress/gutenberg/pull/32038/files#diff-d5f0be6da4449d0e7e208bc29d6ee352d7923f69d26c9c0965a478de0238bb3eR76-R80).